dev/Code/CryEngine/CryPhysics/geometry.h

```diff
@@ -348,7 +349,12 @@ class CGeometry
     virtual Vec3 GetCenter() { return Vec3(ZERO); }
     virtual PhysicsForeignData GetForeignData(int iForeignData = 0) { return iForeignData == m_iForeignData ? m_pForeignData : PhysicsForeignData(0); }
     virtual int GetiForeignData() { return m_iForeignData; }
-    virtual void SetForeignData(PhysicsForeignData pForeignData, int iForeignData) { m_pForeignData = pForeignData; m_iForeignData = iForeignData; }
+	virtual void SetForeignData(PhysicsForeignData pForeignData, int iForeignData)
+	{
+		AZ_Assert(m_iForeignData != DATA_UNSCALED_GEOM, "Overriding foreign data that is reference counted, will leak geometry");
```

Comment:
Added assert to catch the ref-counted pointer being overridden as there is no good way of handling this.